src/sparseml/exporters/transforms/base_transform.py

```diff
@@ -0,0 +1,71 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import ABC, abstractmethod
+from typing import Any
+
+
+class BaseTransform(ABC):
+    """
+    Interface that all transforms must implement.
+    A transform is an object that, when applied to a model,
+    does a single modification to it
+    and returns a modified model.
+    """
+
+    def __call__(self, model: Any) -> Any:
+        return self.apply(model)
+
+    @abstractmethod
+    def apply(self, model: Any) -> Any:
+        """
+        The logic for applying the transform to the model
+        should follow the following steps:
+        1. Validate the input model
+        2. Apply the transform to the model
+        3. Validate the resulting model and return it
+
+        :param model: The input model to be
+            validated, transformed and once again, validated
+        :return: The transformed model
+        """
+        raise NotImplementedError
```

Comment:
optional nit: default implementation
```suggestion
        self.pre_validate(model)
        self.transform(model)
        self.post_validate(model)
```

Reply:
+1

Reply:
additionally, what's the expected flow here? do the validation functions raise errors as they encounter them, or do we return an error string that the apply step will raise?